addons/website_sale/controllers/cart.py

```diff
@@ -409,3 +467,64 @@ def _get_additional_cart_notification_information(self, line):
             infos['uom_name'] = line.product_uom_id.name
 
         return infos
+
+    def quick_reorder(self):
+        sale_orders = request.env['sale.order'].sudo().search(
+                [
+                    ('partner_id', '=', request.env.user.partner_id.id),
+                    ('state', '=', 'sale'),
+                    ('website_id', '=', request.website.id),
+                ],
+                order="date_order desc",
+                limit=10,
+            )
+        cart_lines = list(request.cart.order_line) if request.cart else []
+        today = datetime.today().date()
+        already_accounted_lines = []
+        orders = {}
+
+        def same_combo(line1, line2):
+            return set(line1.linked_line_ids.mapped('product_id').ids) == \
+                set(line2.linked_line_ids.mapped('product_id').ids)
+
+        for order in sale_orders:
+            valid_lines = []
+            for line in order.order_line:
+                product_id = line.product_id.id
+                is_combo = line.product_type == 'combo'
+                if (
+                    line.linked_line_id.product_type == 'combo'
+                    or not line._is_sellable()
+                    or (
+                        request.website.prevent_zero_price_sale
+                        and line.product_id._get_combination_info_variant()['price'] == 0
+                    )
+                ):
+                    continue
+
+                if any(
+                    l.product_id.id == product_id and (not is_combo or same_combo(line, l))
+                    for l in cart_lines + already_accounted_lines
+                ):
+                    continue
+
+                already_accounted_lines.append(line)
+                valid_lines.append(line)
+
+            if not valid_lines:
+                continue
+
+            days_ago = (today - order.date_order.date()).days
+            label = (
+                "Today" if days_ago == 0
+                else "Yesterday" if days_ago == 1
+                else f"{days_ago} days ago"
+            )
+            orders.setdefault(label, []).extend(valid_lines)
+
+        return {
+            'orders': [
```

Comment:
This is not a clear key name when mixed with many other cart-related QWeb values.